src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -15638,5 +15638,213 @@ public void SynthesizedRecordPointerProperty()
             p = comp.GlobalNamespace.GetTypeMember("R").GetMember<SourcePropertySymbolBase>("P3");
             Assert.True(p.HasPointerType);
         }
+
+        [Fact, WorkItem(45008, "https://github.com/dotnet/roslyn/issues/45008")]
+        public void PositionalMemberModifiers_RefOrOut()
+        {
+            var src = @"
+record R(ref int P1, out int P2);
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (2,10): error CS0631: ref and out are not valid in this context
+                // record R(ref int P1, out int P2, in int P3);
+                Diagnostic(ErrorCode.ERR_IllegalRefParam, "ref").WithLocation(2, 10),
+                // (2,22): error CS0631: ref and out are not valid in this context
+                // record R(ref int P1, out int P2, in int P3);
+                Diagnostic(ErrorCode.ERR_IllegalRefParam, "out").WithLocation(2, 22)
+                );
+        }
+
+        [Fact, WorkItem(45008, "https://github.com/dotnet/roslyn/issues/45008")]
+        public void PositionalMemberModifiers_In()
+        {
+            var src = @"
+record R(in int P1);
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+
+            var actualMembers = comp.GetMember<NamedTypeSymbol>("R").GetMembers().ToTestDisplayStrings();
+            var expectedMembers = new[]
+            {
+                "R R.<>Clone()",
```

Comment:
>"R R.<>Clone()", [](start = 16, length = 16)

This is going to break once the name change for the Clone method is merged. In order to avoid unnecessary noise and maintainability burden, consider doing targeted checks only for members that you are interested in.